app/views/filters/_form.html.erb

```diff
@@ -42,9 +42,16 @@
           </span>
       </div>
 
-      <div id="granular_form" style="<%= !f.object.granular? ? 'display:none' : '' %>">
+      <div id="override_taxonomy_form" class="<%= f.object.allows_taxonomies_filtering? && Taxonomy.enabled_taxonomies.any? ? '' : 'hidden' %>">
+        <%= checkbox_f f, :override?,
```

Comment:
Please use line breaks and other stuff to break this up, it's very hard to read as it is now. NIC/auto-suggest IP popover is well done if you want to look at that example

![selection_013](https://cloud.githubusercontent.com/assets/598891/17998033/84d05206-6b72-11e6-8987-4dff8b94e307.png)